src/EnergyEfficientAgent.hpp

```diff
@@ -83,6 +83,7 @@ namespace geopm
             static std::unique_ptr<Agent> make_plugin(void);
             static std::vector<std::string> policy_names(void);
             static std::vector<std::string> sample_names(void);
+            bool do_learning(uint64_t hash, uint64_t hint) const;
```

Comment:
Private? I think the question is, can we get unit test coverage without making this public and adding unit tests for the interface directly?  If there is already good coverage of the logic, seems like it should be private.